docs/fundamentals/code-analysis/style-rules/ide0200.md

```diff
@@ -0,0 +1,79 @@
+---
+title: "IDE0200: Remove unnecessary lambda expression"
+description: "Learn about code analysis rule IDE0200: Remove unnecessary lambda expression"
+ms.date: 07/19/2023
+ms.topic: reference
+f1_keywords:
+- IDE0200
+helpviewer_keywords:
+- IDE0200
+dev_langs:
+- CSharp
+---
+# Remove unnecessary lambda expression (IDE0200)
+
+| Property                 | Value                      |
+| ------------------------ | -------------------------- |
+| **Rule ID**              | IDE0200                    |
+| **Title**                | Remove unnecessary lambda expression |
+| **Category**             | Style                      |
+| **Subcategory**          | Unnecessary code rules     |
+| **Applicable languages** | C# 11+                     |
+
+## Overview
+
+This rule flags the use of a lambda expression where it's unnecessary. Lambda expressions might be unnecessary when the following are all true:
+
+- The expression includes a method invocation.
+- The lambda expression has the same number and order of parameters as the method invocation.
+- The method invocation has no side effects.
+- The lambda expression isn't assigned to a non-delegate type.
+- If the invocation is a generic method, the type arguments are supplied.
+- The invoked method's return type can be converted to the lambda expression's return type.
+
+## Options
+
+This rule has no associated code-style options.
```

Comment:
This rule does have a corresponding code style option: https://github.com/dotnet/roslyn/blob/9999d45981eb74581530c28d297f04ce9ba51004/src/Analyzers/CSharp/Analyzers/RemoveUnnecessaryLambdaExpression/CSharpRemoveUnnecessaryLambdaExpressionDiagnosticAnalyzer.cs#L34
https://github.com/dotnet/roslyn/blob/9999d45981eb74581530c28d297f04ce9ba51004/src/EditorFeatures/Test/Diagnostics/IDEDiagnosticIDConfigurationTests.cs#L861

Though it's purpose is only to enable/disable the rule.